### PR TITLE
add onion protocol

### PR DESCRIPTION
**Pull request by dominictarr**
from @arj03

### CODE REVIEW

**Comment by dominictarr** [comment]
question, @arj03 the "port is not optional" port is the remote port?

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "multiserver",
   "description": "",
-  "version": "1.2.3",
+  "version": "1.2.4",
```
Comment (dominictarr) [inline package.json]: no biggie, but please leave changing the version number of the package to the maintainer

**Comment by dominictarr** [comment]
This looks pretty right, but it means to create a server I have to manually update the tor config for my hidden service, right? I think that's fine for now, but needs some documentation, at least so that I can test it.

**Comment by dominictarr** [comment]
merged into 1.3.0, so we can get going, but we should probably not hardcode the tor port.

**Comment by dominictarr** [comment]
oh, just realized this isn't right at all.
tor uses a socks5 proxy, and tproxy is not that.
https://github.com/AjayMT/tproxy/blob/master/index.js

try this module instead https://github.com/sequoiar/socks5
I don't think we need a server adapter for this protocol yet,
because if you configure tor to proxy your local sbot port,
then the net server will just expose over tor.
later, we can set up a separate server, that say, rejects connections that didn't come through tor.
then if you wish to run ssb over tor _only_ you can do that.

**Comment by dominictarr** [comment]
sorry, that module only implements the server,
this one implements the client:
https://github.com/JoshGlazebrook/socks

**Comment by arj03** [comment]
Alright. I'll whip up a new patch.

Again can't thank you enough for helping me scratch my itch. The next pull request won't be done before I have a connection working ;)

**Pull request merged**